src/model.rs

```diff
@@ -760,6 +760,71 @@ impl Collections {
         self.vehicle_journeys = CollectionWithId::new(vehicle_journeys).unwrap();
     }
 
+    /// Some comments are identical and can be deduplicated
+    pub fn comment_deduplication(&mut self) {
+        let doubloon2ref = &self.get_comment_map_doubloon_to_referent();
+        if doubloon2ref.is_empty() {
+            return;
+        }
+        let doubloons: BTreeSet<String> = doubloon2ref.keys().cloned().collect();
+
+        replace_comment_doubloons_by_ref(&mut self.lines, &doubloons, &doubloon2ref);
+        replace_comment_doubloons_by_ref(&mut self.routes, &doubloons, &doubloon2ref);
+        replace_comment_doubloons_by_ref(&mut self.stop_areas, &doubloons, &doubloon2ref);
+        replace_comment_doubloons_by_ref(&mut self.stop_points, &doubloons, &doubloon2ref);
+        replace_comment_doubloons_by_ref(&mut self.stop_locations, &doubloons, &doubloon2ref);
+
+        fn replace_comment_doubloons_by_ref<T>(
+            collection: &mut CollectionWithId<T>,
+            doubloons: &BTreeSet<String>,
+            doubloons2ref: &BTreeMap<String, String>,
+        ) where
+            T: Id<T> + CommentLinks,
+        {
+            let mut map_pt_object_doubloons: BTreeMap<Idx<T>, Vec<String>> = BTreeMap::new();
+            for (idx, pt_object) in collection.iter() {
+                let intersection: Vec<String> = pt_object
+                    .comment_links()
+                    .intersection(&doubloons)
+                    .cloned()
+                    .collect();
+                if !intersection.is_empty() {
+                    map_pt_object_doubloons.insert(idx, intersection);
+                }
+            }
+
+            for (idx, intersection) in map_pt_object_doubloons {
+                for i in intersection {
+                    let mut pt_object = collection.index_mut(idx);
+                    pt_object.comment_links_mut().remove(&i);
+                    pt_object
+                        .comment_links_mut()
+                        .insert(doubloons2ref[&i].clone());
+                }
+            }
+        }
+    }
+
+    /// From comment collection only, return a map of the similar comments.
+    ///
+    /// Result: doubloons (comments to be removed) are mapped to their similar
+    /// referent (unique to be kept)
+    fn get_comment_map_doubloon_to_referent(&self) -> BTreeMap<String, String> {
+        let mut doubloon2ref: BTreeMap<String, String> = BTreeMap::new();
+        // Map of the referent comments id (uniqueness given the similarity_key)
+        let mut map_ref: BTreeMap<&str, &str> = BTreeMap::new();
+
+        for (_, comment) in &self.comments {
```

Comment:
```suggestion
        let mut map_ref: HashMap<&str, &str> = HashMap::new();

        for comment in self.comments.values() {
```
Hashmap less heavy (and no order needed for map_ref)

Reply:
or `let mut map_ref = HashMap::<&str, &str>::new();` you save 1 `HashMap`